src/Drips.sol

```diff
@@ -471,24 +473,23 @@ library Drips {
                     _currTimestamp(),
                     newDefaultEnd
                 );
-                _moveDeltaRange(
+                _updateDrip(
                     deltas,
                     cycleSecs,
+                    currRecv.amtPerSec,
                     currStart,
-                    currEnd,
                     newStart,
-                    newEnd,
-                    currRecv.amtPerSec
+                    currEnd,
+                    newEnd
                 );
+                currIdx++;
+                newIdx++;
```

Comment:
I think it is easier to read if the increase of both indexes happens here instead of at the bottom of the while loop.